src/main/controllers/helpers/DocumentHelpers.ts

```diff
@@ -75,3 +78,40 @@ export function formatBytes(bytes: number, decimals = 2): string {
 
   return `${parseFloat((bytes / Math.pow(k, i)).toFixed(dm))}${sizes[i]}`;
 }
+
+export function getClaimantResponseDocId(selectedApplication: GenericTseApplicationTypeItem): string {
+  let claimantResponseDocId = undefined;
+  const claimantResponseDoc = getClaimantResponseDocDownload(selectedApplication);
+  if (claimantResponseDoc !== undefined) {
+    claimantResponseDocId = getDocId(claimantResponseDoc.document_url);
+  }
+  return claimantResponseDocId;
+}
+
+export function getDecisionDocId(req: AppRequest, selectedApplication: GenericTseApplicationTypeItem): string {
+  let decisionDocId = undefined;
+  const docId = req.params.docId;
+  const decisionDocIds = [];
+  if (selectedApplication?.value.adminDecision?.length) {
+    const adminDecisions = selectedApplication.value.adminDecision;
+    for (let i = adminDecisions.length - 1; i >= 0; i--) {
+      if (adminDecisions[i].value.responseRequiredDoc !== undefined) {
+        decisionDocIds[i] = adminDecisions[i].value.responseRequiredDoc.document_url;
+      }
+    }
+  }
+  decisionDocId = decisionDocIds.map(it => getDocId(it)).find(id => id === docId);
+  return decisionDocId;
+}
+
+export function getRequestDocId(req: AppRequest): string {
+  const docId = req.params.docId;
+  const userCase = req.session?.userCase;
+  let requestDocId = undefined;
+  if (userCase.selectedRequestOrOrder?.value.sendNotificationUploadDocument?.length) {
+    requestDocId = userCase.selectedRequestOrOrder?.value.sendNotificationUploadDocument
```

Comment:
userCase.selectedRequestOrOrder?.value.sendNotificationUploadDocument? is used twice, could it be put into a variable?

Reply:
Put in variable